libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -565,8 +566,20 @@ private void removeAllMarkers() {
   }
 
   private void updateMapWaynameWithLocation(Location location) {
-    LatLng latLng = new LatLng(location);
-    PointF mapPoint = mapboxMap.getProjection().toScreenLocation(latLng);
-    mapWayname.updateWaynameWithPoint(mapPoint, waynameLayer);
+    if (mapWayname.isAutoQueryEnabled() && onUpdateWaynameListener != null) {
+      onUpdateWaynameListener.updateWaynameWithLocation(location);
+    } else {
+      LatLng latLng = new LatLng(location);
+      PointF mapPoint = mapboxMap.getProjection().toScreenLocation(latLng);
+      mapWayname.updateWaynameWithPoint(mapPoint, waynameLayer);
+    }
+  }
+
+  public void setOnUpdateWaynameListener(OnUpdateWaynameListener onUpdateWayname) {
```

Comment:
Could you add Javadocs here?

Reply:
Also, could you place this `public` method with the rest (before `private` ones)?